build-release.sh

```diff
@@ -5,8 +5,11 @@ set -eEuo pipefail
 eval "$(go env)"
 
 OUT_DIR="${1-dist}"
-# To override the latest git tag as the version, pass something else as the second arg.
-VERSION=${2:-$(git describe --tags --always --dirty)}
+
+if [ -z "$VERSION" ]; then
+    # To override the latest git tag as the version, pass something else as the second arg.
+    VERSION=${2:-$(git describe --tags --always --dirty)}
+fi
```

Comment:
This seems something old, is it still required? The VERSION value should be always set from the `configure` job and in any case, the tags are not available, the checkout action doesn't fetch them.

Reply:
I think it's still valuable if we ever want to run this outside of CI, e.g. locally for some reason.

Reply:
Ok, then I think we can keep it with the `if` condition.

Reply:
Sure, though the `if` is not strictly necessary, since the parameter substitution serves the same purpose if the version is passed as an argument. But it's fine.

Reply:
I am confused by this, `${2:-$(git describe --tags --always --dirty)}` means take the second positional argument or use the result of `git describe ...` if that was empty. Why should we also support an external `VERSION` env var, we are only using positional arguments everywhere else in the script :confused: 

Soft :-1: for this, unless we decide to refactor the whole script, but that should happen in some other PR